dfa_identify/encoding.py

```diff
@@ -44,22 +49,75 @@ class Codec:
     n_nodes: int
     n_colors: int
     n_tokens: int
+    symm_mode: SymmBreak = attr.ib(default = SymmBreak.CLIQUE)
+
+    counts: list[int] = attr.ib()
+    @counts.default
+    def counts_default(self):
+        """
+        Compute number of variables of each type.
+        Used for calculating offsets for decoding and encoding
+        Order: z, x, y, p, t, m
+        """
+        return [
+            self.n_colors,
```

Comment:
Maybe break up into 6 properties for the various counts?

Reply:
It's easier to have it as a tuple, because we need to sum the counts later